server/utils/passport/local-signup.js

```diff
@@ -1,96 +1,89 @@
-import passport from 'passport';
-import config from '../../configs/config';
-import { MongoClient } from 'mongodb';
-import { hash } from '../crypto/hash';
-import { getMongoURI } from '../mongoUtil';
-import defaultUserConfig from '../../configs/defaultUserConfig';
-import basePathConfig from '../../configs/basePathConfig';
+import passport from 'passport'
+import config from '../../configs/config'
+import { hash } from '../crypto/hash'
+import { collections } from '../mongoCollections'
+import basePathConfig from '../../configs/basePathConfig'
+import defaultUserConfig from '../../configs/defaultUserConfig'
 
-const basePath = basePathConfig || '';
+const basePath = basePathConfig || ''
 
 export default (req, res, next) => {
-  passport.authenticate('local-signup', { session: true }, function (err, user, reqBody) {
-    if (err) {
-      return res.redirect(`${basePath}/login?e=${err}`);
-    }
-    if (user) {
-      return res.redirect(`${basePath}/signup?e=existingUser`);
-    }
-    const password = reqBody.password;
-    const username = reqBody.username;
-    const email = reqBody.email;
-    const display_name = reqBody.display_name;
-
-    const mongoURI = getMongoURI({ settings: config.database.mongo });
-    MongoClient.connect(mongoURI, config.database.mongo.server, function (err, client) {
+  passport.authenticate(
+    'local-signup',
+    { session: true },
+    async function (err, user, reqBody) {
       if (err) {
-        console.error(err.message);
-        return res.redirect(`${basePath}/login?e=forbidden`);
+        return res.redirect(`${basePath}/login?e=${err}`)
```

Comment:
do we have route constants for these paths?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/241/commits/2e3f2e5a20221046bc208d7918023e42684da6e2

Reply:
That commit does not resolve this comment.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/241/commits/be3741398c54927adbe373fa7b03098b9ba0424c